### PR TITLE
#1410 - Update images after folder deletion

### PR DESCRIPTION
**Pull request by gabrieldagama**
### Description (*)

Added an event to folder deletion function, extended the masonry grid, and added the reloadGrid method, I believe it is better to components to handle their own state so the suggestion here is to use events instead of injecting parent components to children.

### Fixed Issues (if relevant)

1. magento/adobe-stock-integration#1410: If the Root is not empty, the image state is not updated when the containing folder has been deleted.

### Manual testing scenarios (*)

1. Go to **Content** -> **Media Gallery**
2. Do not select any folder from the folder tree
3. Save an image Preview or Upload an image
4. Create a new folder named 1 for example
5. Select the previously created folder and Sava Preview in it or Upload an image
6. Delete the Previously created folder
7. Create a new folder with the same name (1)
8. Select the created folder.

The image that was previously added to folder 1 should not be there anymore.

### CODE REVIEW

**Comment by gabrieldagama** [comment]
@magento run all tests

**Comment by gabrieldagama** [comment]
@magento run all tests

**Comment by gabrieldagama** [comment]
@magento run all tests

**Comment by coderimus** [comment]
Failed tests results are not related to the PR changes.

**Review by coderimus — DISMISSED** [review]

**Review by chalov-anton — CHANGES_REQUESTED** [review]
:x: QA Failed

The image disappears from the Media Gallery grid. But if you open Adobe Stock, the **Open in media Gallery** button will be displayed instead of **Save Preview**
op_in_med_gall
If you'll create a new folder with the same name that has been deleted, an _Uncaught TypeError: Cannot read property 'replace' of undefined_ will appear
replace

**Comment by gabrieldagama** [comment]
@magento run all tests

File: MediaGalleryUi/view/adminhtml/web/js/grid/masonry.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+            this.initEvents();
+
+            return this;
+        },
+
+        /**
+         * Initialize events
+         */
+        initEvents: function () {
+            $(window).on('enhanced_media_gallery_folder_deleted', this.reloadGrid.bind(this));
```
Comment (sivaschenko) [inline MediaGalleryUi/view/adminhtml/web/js/grid/masonry.js]: Let's rename the event for consistency with other events naming
```suggestion
            $(window).on('folderDeleted.enhancedMediaGallery', this.reloadGrid.bind(this));
```

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Hi @gabrieldagama thanks for the pull request, please see my review comment

**Comment by gabrieldagama** [comment]
@magento run all tests

**Comment by gabrieldagama** [comment]
@magento run all tests

**Review by sivaschenko — DISMISSED** [review]

**Comment by sivaschenko** [comment]
Static tests need to be fixed

**Comment by gabrieldagama** [comment]
@magento run Static Tests

**Comment by gabrieldagama** [comment]
@magento run Static Tests

**Comment by gabrieldagama** [comment]
@magento run all tests

**Comment by gabrieldagama** [comment]
@sivaschenko static tests are fixed!

**Review by sivaschenko — APPROVED** [review]

**Comment by chalov-anton** [comment]
:heavy_check_mark: QA Passed
- the image disappeared from the Media Gallery.
- the **Open in media Gallery** button was changed to **Save Preview**
- also rechecked deleting folder, when there are multiple images.

**Pull request merged**